components/tools/OmeroJava/test/integration/AbstractServerTest.java

```diff
@@ -285,6 +298,155 @@ public void tearDown() throws Exception {
     }
 
     /**
+     * Cast the map of strings (e.g. the ALL_GROUPS_CONTEXT)
+     * into implicit context, which asks for <String, String>
+     * @param implicitContext the implicit context to be changed
+     * @param newContext the map of strings (e.g. ALL_GROUPS_CONTEXT)
+     */
+    protected void mergeIntoContext(ImplicitContext implicitContext, Map<String, String> newContext) {
+        for (final Entry<String, String> entry : newContext.entrySet()) {
+            implicitContext.put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * Add the given annotation to the given image.
+     * @param image an image
+     * @param annotation an annotation
+     * @return the new loaded link from the image to the annotation
+     * @throws ServerError an error possibly occurring during saving of the link
+     */
+    protected ImageAnnotationLink linkImageAnnotation(Image image, Annotation annotation) throws ServerError {
```

Comment:
While personally I am happy and comfortable with using the API directly -- often the gateway brings overhead I don't need, like extra JOINs in its queries -- our training material remains thin enough in comparison with our integration tests that in practice I have used our tests as a significant source of training material. If we want to teach writers of clients to use the gateways then it is probably good if our own integration test code uses them.